app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java

```diff
@@ -1126,7 +1142,20 @@ public void handleResult(@NonNull StreamInfo info) {
             videoUploadDateView.setVisibility(View.GONE);
         }
 
-        prepareDescription(info.getDescription());
+        int serviceId = info.getServiceId();
+
+        if (serviceId != YOUTUBE_SERVICE_ID) {
+            videoDescriptionView.setAutoLinkMask(Linkify.WEB_URLS);
+        }
+
+        if (serviceId == PEERTUBE_SERVICE_ID) {
```

Comment:
Maybe we can avoid the if conditions here by creating a description class in the extractor with fields `content` and `type`.
`prepareDescription(info.getDescription().getContent(), info.getDescription().getType());`

Reply:
Sounds like a good solution

Reply:
Isn't it better for readability to make
``prepareDescription(info.getDescription())``, with ``prepareDescription(Description description)``? But yeah creating a class sounds good.

Reply:
Done, I need more test though. Everything seems fine except YouTube description timestamps. I'll test this weekend.